### ISSUE TITLE
Improve efficiency of HysteresisDeadband::is_in_deadband

### ISSUE THREAD

**Issue by rhaschke** [shadow-robot/sr_core#196]
The average over last errors, this class uses a `std::deque` to maintain a sliding window of previous errors:
https://github.com/shadow-robot/sr_core/blob/9d90e751d58356bf9c9ad38df38142308c7bfab6/sr_utilities/include/sr_utilities/sr_deadband.hpp#L90
https://github.com/shadow-robot/sr_core/blob/9d90e751d58356bf9c9ad38df38142308c7bfab6/sr_utilities/include/sr_utilities/sr_deadband.hpp#L136

While this approach is semantically correct, it is highly inefficient, copying 50 values around in each call. 
There are much more efficient ways to just a compute a sliding average:
- Use a [circular buffer](https://www.codeproject.com/Articles/1185449/Performance-of-a-Circular-Buffer-vs-Vector-Deque-a), maintain the average sum, and just subtract the popped and add the pushed value (all others values remain the same).
- Use estimated sliding average (which should be sufficient here): 
  `avg_error = (1.0 - lambda) * avg_error + lambda * error`
  where lambda adjusts the smoothing and should be chosen `lambda=1.0/nb_errors_for_avg` here.

I strongly suggest the latter method and storing lambda _once_ in the class (`nb_errors_for_avg` shouldn't change over time anyway!) As this requires API changes, I leave it to you to decide on this and implement it.

**Comment by toliver** [comment]
I agree we should make this more efficient. We've also found recently that the default average of 50 values introduces an undesirable zeroing of the error ~25ms after a zero crossing.
We should have expected this, of course. The default averaging window should be much narrower (or not use averaging at all) to get rid of this delay.
